imxrt-hal/src/can/frame.rs

```diff
@@ -0,0 +1,445 @@
+            v if v == FlexCanMailboxCSCode::RxBusy as u8 => FlexCanMailboxCSCode::RxBusy,
+            v if v == FlexCanMailboxCSCode::TxInactive as u8 => FlexCanMailboxCSCode::TxInactive,
+            v if v == FlexCanMailboxCSCode::TxAbort as u8 => FlexCanMailboxCSCode::TxAbort,
+            v if v == FlexCanMailboxCSCode::TxOnce as u8 => FlexCanMailboxCSCode::TxOnce,
+            v if v == FlexCanMailboxCSCode::TxAnswer as u8 => FlexCanMailboxCSCode::TxAnswer,
+            _ => FlexCanMailboxCSCode::Unknown,
+        }
+    }
+}
+
+impl FlexCanMailboxCSCode {
+    #[inline(always)]
+    pub fn to_code_reg(&self) -> u32 {
+        (*self as u32) << CodeReg::CODE_SHIFT
+    }
+
+    #[inline(always)]
+    pub fn from_code_reg(reg: u32) -> Self {
+        Self::from(((reg & CodeReg::CODE_MASK) >> CodeReg::CODE_SHIFT) as u8)
+    }
+
+    #[inline(always)]
+    pub fn is_tx_mailbox(&self) -> bool {
+        (*self as u8) >> 3 != 0
+    }
+}
+
+/// Code Register of a FlexCAN mailbox.
+///
+/// Contains info relating to the frame's remote status (RTR), whether
+/// the frams is standard os extended (IDE), the length of the data
+/// content (DLC), and the timestamp.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct CodeReg(u32);
+
+impl CodeReg {
+    pub(crate) const CODE_SHIFT: u32 = 24;
+    pub(crate) const CODE_MASK: u32 = 0b1111_u32 << Self::CODE_SHIFT;
+
+    #[allow(dead_code)]
+    const SRR_SHIFT: u32 = 22;
+    #[allow(dead_code)]
+    const SRR_MASK: u32 = 0b1_u32 << Self::SRR_SHIFT;
+
+    const IDE_SHIFT: u32 = 21;
+    const IDE_MASK: u32 = 0b1_u32 << Self::IDE_SHIFT;
+
+    const RTR_SHIFT: u32 = 20;
+    const RTR_MASK: u32 = 0b1_u32 << Self::RTR_SHIFT;
+
+    const DLC_SHIFT: u32 = 16;
+    const DLC_MASK: u32 = 0b111_u32 << Self::DLC_SHIFT;
+
+    const TIMESTAMP_SHIFT: u32 = 0;
+    const TIMESTAMP_MASK: u32 = 0xFFFF_u32 << Self::TIMESTAMP_SHIFT;
+
+    /// Creates a new code reg.
+    pub fn new(code: u32) -> Self {
+        Self(code)
+    }
+
+    /// Creates a new code reg for a standard identifier.
+    pub fn new_standard() -> Self {
+        Self::new(0x00)
+    }
+
+    /// Creates a new code reg for an extended identifier.
+    pub fn new_extended() -> Self {
+        Self::new(Self::IDE_MASK)
+    }
+
+    /// Returns the register value.
+    #[inline(always)]
+    pub fn to_code_reg(&self) -> u32 {
+        self.0
+    }
+
+    /// Set the 4 bit code content for a CodeReg
+    #[inline(always)]
+    pub fn set_code(&mut self, code: FlexCanMailboxCSCode) -> Self {
+        Self::new(self.0 & ((code as u32) << Self::CODE_SHIFT))
+    }
+
+    /// Get the 4 bit code content for a CodeReg.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if a matching [`FlexCanMailboxCSCode`] is not found.
+    #[inline(always)]
+    pub fn code(&self) -> FlexCanMailboxCSCode {
+        FlexCanMailboxCSCode::try_from(((self.0 & Self::CODE_MASK) >> Self::CODE_SHIFT) as u8)
+            .unwrap()
+    }
```

Comment:
Now that we implemented `From<u8>` instead of `TryFrom<u8>` for `FlexCanMailboxCSCode`, does this still panic? Could we use `from` and avoid an `unwrap()`?